### PR TITLE
Bug fix longwave rad bep bem

### PR DESCRIPTION
**Pull request by andreazonato**
TYPE: bug fix

KEYWORDS: BEP+BEM, longwave radiation

SOURCE: Andrea Zonato (University of Trento, Italy)

DESCRIPTION OF CHANGES:
There was an error in the previous version on the calculation of long-wave radiation. The variable rl_emit now accounts for 
the emissivity by the photovoltaic panel. Moreover, the variable rld was incorrectly calculated. This will affect the calculation 
of TSK, in urban areas, that in the previous version was wrong.

LIST OF MODIFIED FILES: 
M phys/module_sf_bep_bem.F

TESTS CONDUCTED:
1. This error affects the calculation of TSK, which now is correctly calculated. As you can see in the figure, there is a 
large bias for the TSK of the roofs. Now it is more realistic.
FIGURE_DIFFERENCES
2. Jenkins test are OK.

RELEASE NOTE: There was an error in the previous version on the calculation of long-wave radiation. The variable rl_emit now accounts for the emissivity by the photovoltaic panel. Moreover, the variable rld was incorrectly calculated. This will affect the calculation of TSK in urban areas.

### CODE REVIEW

**Comment by dudhia** [comment]
Need @cenlinhe to review this.
Also include a statement on the size of impacts of this change.

**Comment by cenlinhe** [comment]
The modification looks good to me. @andreazonato As mentioned by Jimy, could you please provide a brief summary and/or a simple figure showing the change in the simulation results due to this modification? For example, which output variables are affected and by how much?

File: phys/module_sf_bep_bem.F [hunk 1]
```diff
@@ -5312,13 +5312,13 @@ subroutine upward_rad(ndu,nzu,ws,bs,sigma,pb,ss,                &
...
          do iz=2,nzu
-            rl_emit=rl_emit-(emr_u*sigma*((1.-pv_frac_roof)*tr_av(id,iz)**4.+pv_frac_roof*tpvlev(id,iz)**4)+ &
-             (1-emr_u)*rld)*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
-            rl_inc=rl_inc+(1-pv_frac_roof*rld+pv_frac_roof*emr_u*sigma*tpvlev(id,iz)**4)*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
-            rs_abs=rs_abs+((1.-albr_u)*rs*(1.-pv_frac_roof))*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
-            gfl=(1.-albr_u)*rs*(1-pv_frac_roof)+emr_u*rld*((1-pv_frac_roof)+pv_frac_roof*emr_u*sigma*tpvlev(id,iz)**4) &
+             rl_emit=rl_emit-(emr_u*sigma*(1.-pv_frac_roof)*tr_av(id,iz)**4.+0.79*sigma*pv_frac_roof*tpvlev(id,iz)**4+ &
+                     (1-emr_u)*rld*(1.-pv_frac_roof)+(1-0.79)*pv_frac_roof*rld)*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
+             rl_inc=rl_inc+rld*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
+             rs_abs=rs_abs+((1.-albr_u)*rs*(1.-pv_frac_roof)+(1.-0.11)*rs*pv_frac_roof)*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
```
Comment (cenlinhe) [inline phys/module_sf_bep_bem.F:5318]: Here, is it assumed that the albedo of pv is 0.11?

File: phys/module_sf_bep_bem.F [hunk 2]
```diff
@@ -5312,13 +5312,13 @@ subroutine upward_rad(ndu,nzu,ws,bs,sigma,pb,ss,                &
          grdflx_urb=grdflx_urb-gfl*ws(id)/(ws(id)+bs(id))/ndu  
  
          do iz=2,nzu
-            rl_emit=rl_emit-(emr_u*sigma*((1.-pv_frac_roof)*tr_av(id,iz)**4.+pv_frac_roof*tpvlev(id,iz)**4)+ &
-             (1-emr_u)*rld)*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
-            rl_inc=rl_inc+(1-pv_frac_roof*rld+pv_frac_roof*emr_u*sigma*tpvlev(id,iz)**4)*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
-            rs_abs=rs_abs+((1.-albr_u)*rs*(1.-pv_frac_roof))*ss(iz)*bs(id)/(ws(id)+bs(id))/ndu
-            gfl=(1.-albr_u)*rs*(1-pv_frac_roof)+emr_u*rld*((1-pv_frac_roof)+pv_frac_roof*emr_u*sigma*tpvlev(id,iz)**4) &
+             rl_emit=rl_emit-(emr_u*sigma*(1.-pv_frac_roof)*tr_av(id,iz)**4.+0.79*sigma*pv_frac_roof*tpvlev(id,iz)**4+ &
```
Comment (cenlinhe) [inline phys/module_sf_bep_bem.F:5315]: In the original code, you assume the same emissivity (emr_u) for both pv and non-pv portion of roof. Now, the modification assumes an emissivity of 0.79 for the pv, is that right?

**Comment by weiwangncar** [comment]
@andreazonato Please add something for Release Note (message).

**Comment by andreazonato** [comment]
I added some explanation. It affects the calculation of TSK, which now is correctly calculated. As you can see in the figure, there is a large bias for the TSK of the roofs. Now it is more realistic
FIGURE_DIFFERENCES

**Review by weiwangncar — APPROVED** [review]

**Review by dudhia — COMMENTED** [review]
I see some questions from Cenlin to address

**Comment by weiwangncar** [comment]
@andreazonato Can you address the comments for the PR at your earliest convenience, if you'd like this PR to go to 4.3.1?

File: phys/module_sf_bep_bem.F (hunk 1 above)
Comment (andreazonato) [reply phys/module_sf_bep_bem.F:5318]: Yes, the albedo of the PVP is 0.11, as in module_sf_bem.F

File: phys/module_sf_bep_bem.F (hunk 2 above)
Comment (andreazonato) [reply phys/module_sf_bep_bem.F:5315]: Yes, the model has been updated. The emissivity of the upper face of the PVP is 0.79 (glazer) while the downward is 0.95 (silicon). Sorry for the delay

**Comment by weiwangncar** [comment]
@cenlinhe Can you comment on if Andrea has addressed your questions, and if you're ok with this PR?

**Comment by cenlinhe** [comment]
Yes, I am OK with Andrea's reply and this PR.

**Comment by weiwangncar** [comment]
@dudhia Can you approve this PR now?

**Review by dudhia — APPROVED** [review]
I see some questions from Cenlin to address

**Comment by dudhia** [comment]
maybe Cenlin can go in and mark the comments as resolved

**Comment by cenlinhe** [comment]
> maybe Cenlin can go in and mark the comments as resolved

@dudhia I did not see the place where I can mark the comments as resolved.

**Comment by dudhia** [comment]
you need to go the the Files Changed part of the PR.

**Comment by cenlinhe** [comment]
@dudhia I went into the Files Changed part, but still did not see the "Resolve" button. I only saw this:
image

**Comment by dudhia** [comment]
OK, I see it, maybe because I am a reviewer. I will mark them resolved.
Jimy

**Pull request merged**